### ISSUE TITLE
nonexistent plugin rules breaks Linter

### ISSUE THREAD

**Issue by ygongdev** [ember-template-lint/ember-template-lint#833]
I'm facing an issue where the rules are not working when a plugin is involved.

```
module.exports = {
  plugins: [
    {
      name: 'someRUles',
      rules: {
        'someRules': require('./template-lint-rules/someRules'),
        ...
      },
    },
  ],

  extends: 'recommended',

  rules: {
    'no-partial': true,
    'no-bare-strings': true,
    },
```

If I remove the plugin, the error are triggered as intended. Any ideas on why this is happening would be appreciated :)

**Comment by ygongdev** [comment]
Ok, there were some `rules` that no longer exists from the plugin. Once those were removed, the `rules` start working again.
I think there should be some sort of error thrown if the rules does not exist?

**Comment by ygongdev** [comment]
So the problem seems to be that we require `template-lintrc` by default, but in the case, where one of the `rules` is trying to import from a nonexistent path, there's no error and it will quietly fail
```
function resolveConfigPath(configPath, pluginPath) {
  var usedPath = path.resolve(process.cwd(), configPath || '.template-lintrc');

  if (pluginPath) {
    //throws exception if not found
    pluginPath = resolve.sync(pluginPath, { basedir: path.dirname(usedPath) });
  }

  let filePath = pluginPath || usedPath;
  try {
    return require(filePath); <- HERE
  } catch (e) {
    let isModuleMissingError = ALLOWED_ERROR_CODES.includes(e.code);
    if (!isModuleMissingError) {
      throw e;
    }
```

**Comment by ygongdev** [comment]
The current workaround seems to be just wrapping the `plugin rules` around a `try catch` so if a invalid `require` is used, it'll actually throw.

**Comment by ygongdev** [comment]
Actually the error is intentionally being swallowed and we return an empty `{}` instead. Is this intended?
```
  let filePath = pluginPath || usedPath;
  try {
    return require(filePath);
  } catch (e) {
    let isModuleMissingError = ALLOWED_ERROR_CODES.includes(e.code);
    if (!isModuleMissingError) {
      throw e;
    }

    if (configPath) {
      throw new Error(
        `The configuration file specified (${configPath}) could not be found. Aborting.`
      );
    } else {
      return {}; <- HERE
    }
  }
```

**Comment by rwjblue** [comment]
Hmm. The thing we were trying to avoid there were the module missing errors specifically thrown when the `filePath` is not found (e.g. you have no `.template-lintrc.js` file). It seems like we are **also** swallowing module missing errors when we find the template config but it itself references missing modules.

Definitely a bug. I think the easiest/best solution here would be to use something like `require.resolve` first on the `filePath` in this `try`/`catch` block, then do the _actual_ `require` outside of the `try`/`catch`.

**Comment by ygongdev** [comment]
Ok now that I know the context, that sounds like a great approach!
I can make a PR for this if you aren't already on it.
